notebooks/tests/test_config.py

```diff
@@ -0,0 +1,105 @@
+import yaml
+from pytest import raises
+from testbook import testbook
+from testbook.exceptions import TestbookRuntimeError
+from uwtools.exceptions import UWConfigError, UWConfigRealizeError
+
+all_cells = [*range(0, 57)]
+err_cells = [13, 17, 45]
+non_err_cells = list(filter(lambda n: n not in err_cells, all_cells))
+
+
+def test_get_config():
+    with open("fixtures/config/get-config.yaml", "r", encoding="utf-8") as f:
+        config1_str = f.read().rstrip()
+        config1_dict = yaml.safe_load(config1_str)
```

Comment:
@christinaholtNOAA Could you elaborate on your concern here? I'm also not sure I'm parsing "UW-supported constructors in a fixture file" correctly -- could you give an example of what you mean.

At first glance, I like this test: It uses the lowest-possible-level mechanism -- `.read()` to get a string version of a file, `yaml.safe_load()` to turn that string into a `dict` -- to establish the "expected" values, then asserts that the more-complicated UW-based techniques demonstrated in the notebook produce equivalent values. Using higher-level UW techniques for the "expected" values introduces the possibility of tautology (testing the same thing on both sides of `==`), like `assert YAMLConfig("a.yaml") == YAMLConfig("a.yaml")` is a tautology, and `assert YAMLConfig("a.yaml") == get_yaml_config("a.yaml")` isn't much better as it's exercising nearly identical code. Of course, all of the above eventually calls some form of `yaml.load()`, but IMO constructing "expected" values in the most basic way possible is sound practice.

But I'm curious if I'm missing something, especially about "UW-supported constructors".

Reply:
I've been hesitant to use the UW API in my testing for the reasons @maddenp-noaa stated, and I'm curious to know more about the concerns here too.

Reply:
Sorry, I often use _constructor_ and _tag_ interchangeably here and probably shouldn't -- _tag_ is the `!int` that shows up in the YAML file, while _constructor_ is the chunk of code that tells _pyyaml_ what to do with the value. If we wanted to use UW-supported tags in our examples, then we'd need to load the YAML file with `get_yaml_config`.

I'm not saying we want to or need to here, but it feels like it could lead to problems where we might want some consistency in our tests later on.